pkg/ingester/ingester.go

```diff
@@ -1673,6 +1738,36 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 	}
 
 	i.tsdbMetrics.setRegistryForUser(userID, tsdbPromReg)
+
+	userDB.ephemeralFactory = func() (*tsdb.Head, error) {
+		// TODO: check user limit for ephemeral series. If it's 0, don't create head and return error.
+
+		headOptions := &tsdb.HeadOptions{
+			ChunkRange:                     i.cfg.EphemeralSeriesRetentionPeriod.Milliseconds(),
+			ChunkDirRoot:                   filepath.Join(udir, "ephemeral_chunks"),
+			ChunkWriteBufferSize:           i.cfg.BlocksStorageConfig.TSDB.HeadChunksWriteBufferSize,
+			ChunkEndTimeVariance:           0,
+			ChunkWriteQueueSize:            i.cfg.BlocksStorageConfig.TSDB.HeadChunksWriteQueueSize,
+			StripeSize:                     i.cfg.BlocksStorageConfig.TSDB.StripeSize,
+			SeriesCallback:                 nil, // TODO: handle limits.
+			EnableExemplarStorage:          false,
+			EnableMemorySnapshotOnShutdown: false,
+			IsolationDisabled:              true,
+		}
+
+		// We need to set this, despite OOO time window being 0.
```

Comment:
Can you explain why we need to set it to `i.cfg.BlocksStorageConfig.TSDB.OutOfOrderCapacityMax`?

Reply:
We need *some* value, because Head checks that it's set, but it won't be used as we don't enable OOO. I used `i.cfg.BlocksStorageConfig.TSDB.OutOfOrderCapacityMax` to get some valid value. I will switch it to `tsdb.DefaultOutOfOrderCapMax`.